crates/matrix-sdk-base/src/store/memory_store.rs

```diff
@@ -700,17 +702,44 @@ impl StateStore for MemoryStore {
         Ok(self.custom.write().unwrap().remove(key))
     }
 
-    // The in-memory store doesn't cache media
-    async fn add_media_content(&self, _request: &MediaRequest, _data: Vec<u8>) -> Result<()> {
+    async fn add_media_content(&self, request: &MediaRequest, data: Vec<u8>) -> Result<()> {
+        self.media.write().unwrap().push((request.unique_key(), data));
+
         Ok(())
     }
-    async fn get_media_content(&self, _request: &MediaRequest) -> Result<Option<Vec<u8>>> {
+
+    async fn get_media_content(&self, request: &MediaRequest) -> Result<Option<Vec<u8>>> {
+        let media = self.media.read().unwrap();
+        let expected_key = request.unique_key();
+
+        for (media_key, media_content) in media.iter() {
+            if media_key == &expected_key {
+                return Ok(Some(media_content.to_owned()));
+            }
+        }
+
         Ok(None)
     }
-    async fn remove_media_content(&self, _request: &MediaRequest) -> Result<()> {
+
+    async fn remove_media_content(&self, request: &MediaRequest) -> Result<()> {
+        // Pop all media until the one represented by `request` is found.
+
+        let mut media = self.media.write().unwrap();
+        let expected_key = request.unique_key();
+        let Some(index) = media.iter().position(|(media_key, _)| media_key == &expected_key) else {
+            return Ok(());
+        };
+
+        let _pop = media.drain(0..=index);
```

Comment:
Could we have a `remove` function in the `RingBuffer` instead?

Reply:
Not in this PR :-p.

Reply:
Done.